platform/linux-generic/odp_pool.c

```diff
@@ -722,31 +717,43 @@ int buffer_alloc_multi(pool_t *pool, odp_buffer_hdr_t *buf_hdr[], int max_num)
 	if (odp_unlikely(num_deq)) {
 		/* Temporary copy needed since odp_buffer_t is uintptr_t
 		 * and not uint32_t. */
-		uint32_t data[burst];
+		uint32_t data[CACHE_BURST];
+		uint32_t burst = 0;
+		uint32_t deq = 0;
+		uint32_t mask;
+		ring_t *ring;
 
 		ring      = &pool->ring->hdr;
 		mask      = pool->ring_mask;
-		burst     = ring_deq_multi(ring, mask, data, burst);
-		cache_num = burst - num_deq;
 
-		if (odp_unlikely(burst < num_deq)) {
-			num_deq   = burst;
-			cache_num = 0;
-		}
+		while (num_deq) {
+			burst = ring_deq_multi(ring, mask, data, CACHE_BURST);
+			if (deq > burst)
+				deq = burst;
+			else
+				deq = num_deq;
+
+			for (i = 0; i < deq; i++) {
+				hdr = buf_hdr_from_index(pool, data[i]);
+				odp_prefetch(hdr);
+				buf_hdr[num_ch++] = hdr;
+			}
 
-		for (i = 0; i < num_deq; i++) {
-			uint32_t idx = num_ch + i;
+			num_deq -= deq;
 
-			hdr = buf_hdr_from_index(pool, data[i]);
-			odp_prefetch(hdr);
-			buf_hdr[idx] = hdr;
+			/* ring exhausted */
+			if (odp_unlikely(burst < CACHE_BURST)) {
+				/* No more buffers to dequeue. */
+				num_deq = 0;
```

Comment:
Looking at this again, since `num_deq` is not referenced further in this routine this shouldn't be necessary.